DataDog/datadog-agent
# Report pod annotation AD errors in agent status

### What does this PR do?

Display autodiscovery errors about pod annotations in the agent's `status` command:

```
=============
Autodiscovery
=============

  Errors
  ======
    
    default/nginx-5fbdd9d99b-lxwx9
    ------------------------------
        annotation ad.datadoghq.com/nginx.logs is invalid: nginx doesn't match a container identifier
    
    default/redis-795f59595c-9kjzd
    ------------------------------
        annotation ad.datadoghq.com/redis2.check_names is invalid: redis2 doesn't match a container identifier
        annotation ad.datadoghq.com/redis2.init_configs is invalid: redis2 doesn't match a container identifier
        annotation ad.datadoghq.com/redis2.instances is invalid: redis2 doesn't match a container identifier
        could not extract checks config: in instances: failed to unmarshal JSON: invalid character ']' after top-level value
    
    test/redis-795f59595c-549qj
    ---------------------------
        annotation ad.datadoghq.com/redis2.check_names is invalid: redis2 doesn't match a container identifier [redis]
        annotation ad.datadoghq.com/redis2.init_configs is invalid: redis2 doesn't match a container identifier [redis]
        annotation ad.datadoghq.com/redis2.instances is invalid: redis2 doesn't match a container identifier [redis]
    
```


### Motivation

Feature request

### Additional Notes

This only checks annotations using the new autodiscovery annotation format: `ad.datadoghq.com/`. Legacy `service-discovery.datadoghq.com/` annotations should be ignored.

### Describe your test plan

* Create a pod(s) with incorrect pod annotations, e.g. invalid JSON syntax, container identifiers not matching, missing a required AD annotation like `ad.datadoghq.com/<identifier>.instances`, etc.
* Check the agent status in the agent container: `kubectl exec -it <agent-pod> -c agent -- agent status`
* Changes from deleting or editing pods should be reflected in the agent status
* If there are no errors, the `Autodiscovery` section shouldn't be present in the agent status
